src/CanonicalizeGPUVars.cpp

```diff
@@ -29,51 +29,42 @@ string get_block_name(int index) {
 }
 
 class CountGPUBlocksThreads : public IRVisitor {
-    string prefix;  // Producer name + stage
-
     using IRVisitor::visit;
 
     void visit(const For *op) override {
-        if (starts_with(op->name, prefix)) {
-            if (op->for_type == ForType::GPUBlock) {
-                nblocks++;
-            } else if (op->for_type == ForType::GPUThread) {
-                nthreads++;
-            } else if (op->for_type == ForType::GPULane) {
-                nlanes++;
-            }
-        }
+        // Figure out how much to increment each counter by based on the loop
+        // type.
+        int db = op->for_type == ForType::GPUBlock;
+        int dl = op->for_type == ForType::GPULane;
+        int dt = op->for_type == ForType::GPUThread;
+
+        // The threads counter includes lanes loops
+        dt += dl;
+
+        // Increment counters
+        nb += db;
+        nl += dl;
+        nt += dt;
+
+        // Update the high water mark.
```

Comment:
Is "high water mark" a known term? I have no idea what that means.

Reply:
https://en.wikipedia.org/wiki/High_water_mark
https://encyclopedia2.thefreedictionary.com/high-water+mark
https://stackoverflow.com/questions/45489405/what-are-high-and-low-water-marks-in-bit-streaming

etc